### PR TITLE
PS-480 Generate crux API used by engine

### PR DESCRIPTION
**Pull request by codingrhythm**
This PR implements a new code generator that generates a new s12 API for each service method that will be used by crux engine.

The code it generates will be used by crux engine API from this PR https://github.com/SafetyCulture/crux/pull/1090

### CODE REVIEW

File: protobuf/protoc-gen-cruxclient/api_generator.cc [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+  std::map<string, string> vars;
+  vars["filename_identifier"] = FilenameIdentifier(file->name());
+
+  PrintPrologue(printer, file);
+}
+
+std::string APIGenerator::GetMethodSignature(
+  const std::string& service_name,
+  const std::string& method_name) const {
+  return "k" + service_name + method_name;
```
Comment (rogchap) [inline protobuf/protoc-gen-cruxclient/api_generator.cc]: Never a fan of 1 line functions; can this be in-lined?

Comment (codingrhythm) [reply protobuf/protoc-gen-cruxclient/api_generator.cc]: Actually, I can delete this.

File: protobuf/protoc-gen-cruxclient/api_generator.cc [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    for (int method_index = 0; method_index < service->method_count();
+        ++method_index) {
+      const MethodDescriptor *method = service->method(method_index);
+      vars["method_name"] = method->name();
+      vars["request"] = ClassName(method->input_type(), true);
+      vars["response"] = ClassName(method->output_type(), true);
+      vars["api_name"] = method->name() + "API";
+
+      if (method->client_streaming()) {
+        // [RC]: Client Steaming not supported yet
```
Comment (rogchap) [inline protobuf/protoc-gen-cruxclient/api_generator.cc]: lol

File: protobuf/protoc-gen-cruxclient/api_generator.cc (hunk 1 above)
Comment (codingrhythm) [reply protobuf/protoc-gen-cruxclient/api_generator.cc]: Updated

**Pull request merged**